trino/trino.go

```diff
@@ -1215,22 +1223,34 @@ func (qr *driverRows) Next(dest []driver.Value) error {
 			return err
 		}
 	}
+
 	if len(qr.coltype) == 0 {
 		qr.err = sql.ErrNoRows
 		return qr.err
 	}
-	for i, v := range qr.coltype {
-		if i > len(dest)-1 {
+
+	row := qr.data[qr.rowindex]
```

Comment:
Move any refactor not directly related to the issue this PR fixes to a separate commit.